src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/Formatting/FormattingTestBase.cs

```diff
@@ -183,16 +185,43 @@ private static (RazorCodeDocument, DocumentSnapshot) CreateCodeDocumentAndSnapsh
         {
             fileKind ??= FileKinds.Component;
             tagHelpers ??= Array.Empty<TagHelperDescriptor>();
+            if (fileKind == FileKinds.Component)
+            {
+                tagHelpers = tagHelpers.Concat(GetDefaultRuntimeComponents()).ToArray();
+            }
             var sourceDocument = text.GetRazorSourceDocument(path, path);
+
+            // Yes I know "BlazorServer_31 is weird, but thats what is in the taghelpers.json file
+            const string defaultImports = @"
+@using BlazorServer_31
+@using BlazorServer_31.Pages
+@using BlazorServer_31.Shared
+@using Microsoft.AspNetCore.Components
+@using Microsoft.AspNetCore.Components.Authorization
+@using Microsoft.AspNetCore.Components.Routing
+@using Microsoft.AspNetCore.Components.Web
+";
+
+            var importsPath = new Uri("file:///path/to/_Imports.razor").AbsolutePath;
+            var importsSourceText = SourceText.From(defaultImports);
+            var importsDocument = importsSourceText.GetRazorSourceDocument(importsPath, importsPath);
+            var importsSnapshot = new Mock<DocumentSnapshot>(MockBehavior.Strict);
+            importsSnapshot.Setup(d => d.GetTextAsync()).Returns(Task.FromResult(importsSourceText));
+            importsSnapshot.Setup(d => d.FilePath).Returns(importsPath);
+            importsSnapshot.Setup(d => d.TargetPath).Returns(importsPath);
+
             var projectEngine = RazorProjectEngine.Create(builder =>
             {
                 builder.SetRootNamespace("Test");
                 RazorExtensions.Register(builder);
             });
-            var codeDocument = projectEngine.ProcessDesignTime(sourceDocument, fileKind, Array.Empty<RazorSourceDocument>(), tagHelpers);
+            var codeDocument = projectEngine.ProcessDesignTime(sourceDocument, fileKind, new[] { importsDocument }, tagHelpers);
+
+            Assert.False(codeDocument.GetCSharpDocument().Diagnostics.Any(), "Error creating document: " + string.Join(Environment.NewLine, codeDocument.GetCSharpDocument().Diagnostics));
```

Comment:
it's a shame `Assert.Empty` doesn't have a message right?

Reply:
XUnit is awesome until you have an opinion that Brad disagrees with :)